### DISCUSSION TITLE
using token from oauth from github to make github api requests?

### DISCUSSION THREAD

**Discussion by cannap** [supabase/supabase#397] [General]
Hi, i have a Question atm i never used Supabase but i want!.

So after reading the Auth part is see there i easy implemenation of github oauth but how can i pass a scope and use the auth code for github api request?

**Comment by kiwicopple** [comment]
Hey @cannap !

I think you want to make calls to the GitHub API right? In that case you can just do it without Supabase. You may get rate-limited, so you just need to go here and generate an access token: https://github.com/settings/tokens

Let me know if I misunderstood your question

**Reply by elemdos** [reply]
@kiwicopple Are they any plans to add a 'scope' param to the authentication request? At the moment I'm having to authenticate once with Supabase and another time the old fashioned way to get the token. Firebase for reference: https://firebase.google.com/docs/auth/web/github-auth

**Reply by elemdos** [reply]
Actually, that doesn't seem to be working either. I tried following [repository.surf's example](https://github.com/supabase/repository.surf/blob/10c3a52f3ce004d596f641428f01fd454b35bc00/lib/auth.js#L48) for adding permissions but I'm hitting a wall that it's somehow avoiding: 
![image](https://user-images.githubusercontent.com/39444813/108744871-1d506a00-7543-11eb-9850-e9714d017ecd.png)

Since we need repo permissions, this takes signing in with Github off the table for us

**Reply by kiwicopple** [reply]
> Since we need repo permissions

This one needs to be set up via the GitHub app. When you create a new GitHub app/client, it should ask you what scope/permissions you will ask the user for. 

> Are they any plans to add a 'scope' param to the authentication request

 I'll have to dig into [GoTrue](https://github.com/netlify/gotrue), if it's implemented there it should be easy enough, otherwise we will have to add it to our [fork](https://github.com/supabase/gotrue)

**Reply by elemdos** [reply]
> This one needs to be set up via the GitHub app

We're using an OAuth app (to do actions on behalf of the user), so scopes need to be listed in the request. 

Looks like the current scopes are [hard-coded](https://github.com/supabase/gotrue/blob/4bb48144368a5636abd05c6f0078382968eb10c0/api/provider/github.go#L57). I'll go ahead and add an issue over there. Fingers crossed it can be added soon so we can resume our migration; loving Supabase so far.

**Comment by surjithctly** [comment]
@kiwicopple  Does Supabase store Github Access Token? 

For example, a user signed up with Github, GitHub provides an access token for that user, then we will be able to call GitHub API and do mutation as the user. 

eg: User A signed up with token, then we can create an issue/comment on behalf of that user. For that we need to pass the `access_token`, So just wondering can we get that token stored and retrieve later?

**Reply by pmespresso** [reply]
has there been an answer to this yet?

**Reply by pmespresso** [reply]
hey there me from literally 2 minutes ago, the docs had the solution: `const { data, error } = await supabase.auth.getSession()`

**Comment by jacobhq** [comment]
In case this issue is not already resolved, please see:

- https://supabase.io/docs/reference/javascript/auth-signin#sign-in-with-scopes (See code block)
-  https://twitter.com/supabase/status/1369647270388523010 (Link in tweet does not seem to speak about scopes anymore)

Good luck!

**Comment by stephenhandley** [comment]
lol
